src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleRequestSender.cs

```diff
@@ -337,6 +470,60 @@ private void SendMessageAndListenAndReportTestCases(IEnumerable<string> sources,
             this.testPlatformEventSource.TranslationLayerDiscoveryStop();
         }
 
+        private async Task SendMessageAndListenAndReportTestCasesAsync(IEnumerable<string> sources, string runSettings, ITestDiscoveryEventsHandler eventHandler)
+        {
+            try
+            {
+                this.communicationManager.SendMessage(
+                            MessageType.StartDiscovery,
+                            new DiscoveryRequestPayload() { Sources = sources, RunSettings = runSettings },
+                            this.protocolVersion);
+                var isDiscoveryComplete = false;
+
+                // Cycle through the messages that the vstest.console sends.
+                // Currently each of the operations are not separate tasks since they should not each take much time.
+                // This is just a notification.
+                while (!isDiscoveryComplete)
```

Comment:
This while loop based implementation will eventually get replaced with a event based one. For example, we can compare: [TestRequestSender](https://github.com/Microsoft/vstest/blob/master/src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs#L141) with the newer [TestRequestSender2](https://github.com/Microsoft/vstest/blob/master/src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender2.cs#L180).